src/IssueViz/Models/QuickFixVisualization.cs

```diff
@@ -29,38 +32,69 @@ public interface IQuickFixVisualization
         IQuickFix Fix { get; }
 
         IReadOnlyList<IQuickFixEditVisualization> EditVisualizations { get; }
+
+        bool IsApplicable(ITextSnapshot currentSnapshot);
     }
 
     internal class QuickFixVisualization : IQuickFixVisualization
     {
+        private readonly ISpanTranslator spanTranslator;
+
         public QuickFixVisualization(IQuickFix fix, IReadOnlyList<IQuickFixEditVisualization> editVisualizations)
+            : this(fix, editVisualizations, new SpanTranslator())
+        {
+            Fix = fix;
+            EditVisualizations = editVisualizations;
+        }
+
+        internal QuickFixVisualization(IQuickFix fix,
+            IReadOnlyList<IQuickFixEditVisualization> editVisualizations,
+            ISpanTranslator spanTranslator)
         {
+            this.spanTranslator = spanTranslator;
             Fix = fix;
             EditVisualizations = editVisualizations;
         }
 
         public IQuickFix Fix { get; }
 
         public IReadOnlyList<IQuickFixEditVisualization> EditVisualizations { get; }
+
+        public bool IsApplicable(ITextSnapshot currentSnapshot) =>
+            EditVisualizations.All(x => IsApplicable(x, currentSnapshot));
+
+        private bool IsApplicable(IQuickFixEditVisualization editViz, ITextSnapshot currentSnapshot)
+        {
+            var updatedSpan = spanTranslator.TranslateTo(editViz.Span, currentSnapshot, SpanTrackingMode.EdgeExclusive);
+            var currentText = currentSnapshot.GetText(updatedSpan);
+
+            return string.Equals(currentText, editViz.OriginalText, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
simple text comparison for now

Reply:
Why are we ignoring the case? If the case sensitive change in a class name it could invalidate the span.

Reply:
You're right 👍 I'll fix